basestar-schema/src/main/java/io/basestar/schema/InstanceSchema.java

```diff
@@ -122,20 +124,34 @@ default boolean hasMutableProperties() {
 
     default InstanceSchema resolveExtended(final Name name) {
 
-        if(getQualifiedName().equals(name)) {
+        if (getQualifiedName().equals(name)) {
             return this;
         } else {
             throw new IllegalStateException(name + " is not a valid subtype of " + getQualifiedName());
         }
     }
 
+    default void validateFieldNames(List<String> names) {
+        List<String> duplicates = names.stream()
+                .filter(key -> names.stream()
+                        .filter(target -> !key.equals(target))
+                        .map(Casing.LOWERCASE_SNAKE::name)
+                        .anyMatch(target -> Casing.LOWERCASE_SNAKE.name(key).equals(target)))
+                .collect(Collectors.toList());
```

Comment:
could be done in a single-pass (instead of each-vs-each checking) by doing `names.stream().collect(groupingBy(Casing.LOWERCASE_SNAKE::name))` and checking if any of the entries have >1 value (grouped name with same lowercase representation). Then the output message could group conflicting names.